incubator/exporter-otlp/src/main/java/io/aklivity/zilla/runtime/exporter/otlp/internal/OltpExporterHandler.java

```diff
@@ -0,0 +1,175 @@
+ * License at
+ *
+ *   https://www.aklivity.io/aklivity-community-license/
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.aklivity.zilla.runtime.exporter.otlp.internal;
+
+import static io.aklivity.zilla.runtime.exporter.otlp.internal.config.OtlpSignalsConfig.Signals.METRICS;
+
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.LongFunction;
+
+import io.aklivity.zilla.runtime.engine.EngineConfiguration;
+import io.aklivity.zilla.runtime.engine.EngineContext;
+import io.aklivity.zilla.runtime.engine.config.AttributeConfig;
+import io.aklivity.zilla.runtime.engine.config.KindConfig;
+import io.aklivity.zilla.runtime.engine.exporter.ExporterHandler;
+import io.aklivity.zilla.runtime.engine.metrics.Collector;
+import io.aklivity.zilla.runtime.engine.metrics.reader.MetricsReader;
+import io.aklivity.zilla.runtime.exporter.otlp.internal.config.OtlpEndpointConfig;
+import io.aklivity.zilla.runtime.exporter.otlp.internal.config.OtlpExporterConfig;
+import io.aklivity.zilla.runtime.exporter.otlp.internal.config.OtlpSignalsConfig;
+import io.aklivity.zilla.runtime.exporter.otlp.internal.serializer.OtlpMetricsSerializer;
+
+public class OltpExporterHandler implements ExporterHandler
+{
+    private static final long RETRY_INTERVAL = Duration.ofSeconds(5).toMillis();
+    private static final long WARNING_INTERVAL = Duration.ofMinutes(5).toMillis();
+    private static final String DEFAULT_METRICS_URI = "/v1/metrics";
+    private static final String HTTP = "http";
+
+    private final EngineContext context;
+    private final Set<OtlpSignalsConfig.Signals> signals;
+    private final String protocol;
+    private final URI metricsUrl;
+    private final long interval;
+    private final Collector collector;
+    private final LongFunction<KindConfig> resolveKind;
+    private final List<AttributeConfig> attributes;
+    private final HttpClient httpClient;
+
+    private OtlpMetricsSerializer serializer;
+    private long lastAttempt;
+    private long lastSuccess;
+    private boolean warning;
+
+    public OltpExporterHandler(
+        EngineConfiguration config,
+        EngineContext context,
+        OtlpExporterConfig exporter,
+        Collector collector,
+        LongFunction<KindConfig> resolveKind,
+        List<AttributeConfig> attributes)
+    {
+        this.context = context;
+        this.metricsUrl = createMetricsUrl(exporter.options().endpoint);
```

Comment:
This logic should move naturally to `OtlpEndpointConfig` when we have `URI` config fields and `resolveXXX` methods, such as `resolveMetrics()`.

Reply:
fixed